File: app/src/main/java/ru/alexsas/mywardrobe_kt/screens/main/auth/RegisterFragment.java
package ru.alexsas.mywardrobe.screens.main.auth;

import static android.content.ContentValues.TAG;

import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.navigation.Navigation;
import androidx.navigation.fragment.NavHostFragment;

import android.text.TextUtils;
import android.util.Log;
import android.util.Patterns;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

import ru.alexsas.mywardrobe.databinding.FragmentRegisterBinding;
import ru.alexsas.mywardrobe.R;


public class RegisterFragment extends Fragment {

    private FirebaseAuth mAuth;
    private FragmentRegisterBinding mBinding;


    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        mBinding = FragmentRegisterBinding.inflate(inflater, container, false);
        return mBinding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        //Buttons

        mBinding.registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                String email = mBinding.emailEditText.getText().toString();
                String password = mBinding.passwordEditText.getText().toString();
                String reply_password = mBinding.retryPasswordEditText.getText().toString();
                createAccount(email, password, reply_password);
            }
        });

        mBinding.cancelButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Navigation.findNavController(view).popBackStack();
            }
        });


        mAuth = FirebaseAuth.getInstance();

    }


    private boolean validateForm() {
        boolean valid = true;

        String email = mBinding.emailEditText.getText().toString();
        if (TextUtils.isEmpty(email) && !Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            mBinding.emailTextInput.setError(getString(R.string.email_error_msg));
            valid = false;
        } else {
            mBinding.emailTextInput.setError(null);
        }

        String password = mBinding.passwordEditText.getText().toString();
        if (TextUtils.isEmpty(password) || password.length() <= 8) {
            mBinding.passwordTextInput.setError(getString(R.string.password_error_msg));
            valid = false;
        } else {
            mBinding.passwordTextInput.setError(null);
        }

        String retry_password = mBinding.retryPasswordEditText.getText().toString();
        if (!password.equals(retry_password)) {
            mBinding.retryPasswordTextInput.setError(getString(R.string.retry_password_error_msg));
            valid = false;
        } else {
            mBinding.retryPasswordTextInput.setError(null);
        }

        return valid;
    }

    private void createAccount(String email, String password, String retry_passrord) {
        Log.d(TAG, "createAccount:" + email);

        if (!validateForm()) {
            return;
        }


        mAuth.createUserWithEmailAndPassword(email, password)
                .addOnCompleteListener(requireActivity(), new OnCompleteListener<AuthResult>() {
                    @Override
                    public void onComplete(@NonNull Task<AuthResult> task) {
                        if (task.isSuccessful()) {
                            // Sign in success, update UI with the signed-in user's information
                            Log.d(TAG, "createUserWithEmail:success");
                            FirebaseUser user = mAuth.getCurrentUser();
                            Toast.makeText(getContext(), "Account successfully created!",
                                    Toast.LENGTH_SHORT).show();
                        } else {
                            // If sign in fails, display a message to the user.
                            Log.w(TAG, "createUserWithEmail:failure", task.getException());
                            Toast.makeText(getContext(), "Account creation failed",
                                    Toast.LENGTH_SHORT).show();
                        }

                    }
                });
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        mBinding = null;
    }

}

File: app/src/main/java/ru/alexsas/mywardrobe_kt/screens/main/auth/RegisterFragment.kt
package ru.alexsas.mywardrobe_kt.screens.main.auth

import android.content.ContentValues
import android.os.Bundle
import android.text.TextUtils
import android.util.Log
import android.util.Patterns
import android.view.View
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.navigation.Navigation.findNavController
import androidx.navigation.fragment.findNavController
import com.google.firebase.auth.FirebaseAuth
import ru.alexsas.mywardrobe_kt.R
import ru.alexsas.mywardrobe_kt.databinding.FragmentRegisterBinding

@Suppress("NAME_SHADOWING")
class RegisterFragment: Fragment(R.layout.fragment_register) {

    private lateinit var mAuth: FirebaseAuth
    private lateinit var mBinding: FragmentRegisterBinding




    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        mBinding = FragmentRegisterBinding.bind(view)
        //Buttons
        mBinding.registerButton.setOnClickListener {
            val email: String = mBinding.emailEditText.text.toString()
            val password: String = mBinding.passwordEditText.text.toString()
            mBinding.retryPasswordEditText.text.toString()
            createAccount(email, password)
        }
        mBinding.cancelButton.setOnClickListener { view ->
            findNavController(
                view
            ).popBackStack()
        }
        mAuth = FirebaseAuth.getInstance()
    }


    private fun validateForm(): Boolean {
        var valid = true
        val email: String = mBinding.emailEditText.text.toString()
        if (TextUtils.isEmpty(email) || !Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
            mBinding.emailTextInput.error = getString(R.string.email_error_msg)
            valid = false
        } else {
            mBinding.emailTextInput.error = null
        }
        val password: String = mBinding.passwordEditText.text.toString()
        if (TextUtils.isEmpty(password) || password.length <= 8) {
            mBinding.passwordTextInput.error = getString(R.string.password_error_msg)
            valid = false
        } else {
            mBinding.passwordTextInput.error = null
        }
        val retry_password: String = mBinding.retryPasswordEditText.text.toString()
        if (password != retry_password) {
            mBinding.retryPasswordTextInput.error = getString(R.string.retry_password_error_msg)
            valid = false
        } else {
            mBinding.retryPasswordTextInput.error = null
        }
        return valid
    }

    private fun createAccount(email: String, password: String) {
        Log.d(ContentValues.TAG, "createAccount:$email")
        if (!validateForm()) {
            return
        }
        mAuth.createUserWithEmailAndPassword(email, password)
            .addOnCompleteListener(
                requireActivity()
            ) { task ->
                if (task.isSuccessful) {
                    // Sign in success, update UI with the signed-in user's information
                    Log.d(ContentValues.TAG, "createUserWithEmail:success")
                    Toast.makeText(
                        context, "Account successfully created!",
                        Toast.LENGTH_SHORT
                    ).show()
                    findNavController().navigate(R.id.action_registerFragment_to_tabsFragment)
                } else {
                    // If sign in fails, display a message to the user.
                    Log.w(ContentValues.TAG, "createUserWithEmail:failure", task.exception)
                    Toast.makeText(
                        context, "Account creation failed",
                        Toast.LENGTH_SHORT
                    ).show()
                }
            }
    }

}